### DISCUSSION TITLE
Issue regarding memory consumption while running bonsai workflow to record movie from Ximea camera

### DISCUSSION THREAD

**Discussion by arpi2203** [bonsai-rx/bonsai#1690] [Q&A]
Hi
I am running a Bonsai code in Visual studio which activates a bonsai workflow. When that workflow is started, it record movie from a ximea camera and compress the movie and save it in a desktop folder. I want to record a movie of 4 hours or more but when that bonsai workflow is running, it consumes a lot of memory and the memory consumption keeps on increasing and finally the computation fails in less than 1 hour. Could anybody please help me how to resolve it?
 The original bonsai code which activates the workflow is given in a word document namely "Bonsai.Ximea_main.docx".
The whole workflow is given in the attached ppt. Other figures are attched regarding how the visual studio with bonsai code looks and how the workflow looks.
The system specification is-64 bits operating system with 128GB RAM.
Thanks

![Bonsai in VS](https://github.com/bonsai-rx/bonsai/assets/160875606/4badd528-cd9f-4cba-8cde-ad8405be7d61)
![systemm info](https://github.com/bonsai-rx/bonsai/assets/160875606/97dff63a-76d0-444a-92b3-9dba9cd26cdf)
![workflow1](https://github.com/bonsai-rx/bonsai/assets/160875606/4760b880-dbd7-4773-86b6-dd1e1c461f10)
![workflow2](https://github.com/bonsai-rx/bonsai/assets/160875606/1a80d313-d62f-4b17-9180-186c32c3b89c)

[Bonsai workflow.pptx](https://github.com/bonsai-rx/bonsai/files/14395537/Bonsai.workflow.pptx)

[Bonsai.Ximea_main.docx](https://github.com/bonsai-rx/bonsai/files/14395535/Bonsai.Ximea_main.docx)

**Comment by bruno-f-cruz** [comment]
Hi
Please attach your bonsai workflow. As I already said in your other 2 posts, this is likely a problem with the codec you are using and the PC not keeping up with the encoding. Find a more performant codec or save images as raw, decrease frame rate or resolution. These are all valid solutions. 
Once you add the workflow file I am happy to take a look.
